staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -56,7 +57,20 @@ func newMutatingDispatcher(p *Plugin) func(cm *webhook.ClientManager) generic.Di
 var _ generic.Dispatcher = &mutatingDispatcher{}
 
 func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr *generic.VersionedAttributes, o admission.ObjectInterfaces, relevantHooks []*v1beta1.Webhook) error {
+	reinvokeCtx := attr.GetReinvocationContext()
+	if reinvokeCtx.IsReinvoke() && reinvokeCtx.IsOutputChangedSinceLastWebhookInvocation(attr.Attributes.GetObject()) {
+		// If the object has changed, we know the in-tree plugin re-invocations have mutated the object,
+		// and we need to reinvoke all eligible webhooks.
+		reinvokeCtx.RequireReinvokingPreviouslyInvokedPlugins()
```

Comment:
And if the in-tree plugins do not change anything? We won't call the webhooks anymore? Why?

Reply:
we won't flush the `IfNeeded` buffer to the `Needed` set if the in-tree plugins do not change anything

there can be webhooks in the `Needed` set already if the buffer has been flushed in the first pass

Reply:
@sttts - See discussion in https://github.com/kubernetes/kubernetes/pull/78080#discussion_r286128070